app/controllers/api/v1/accounts_controller.rb

```diff
@@ -47,6 +47,8 @@ def follow
     options = @account.locked? || current_user.account.silenced? ? {} : { following_map: { @account.id => { reblogs: follow.show_reblogs?, notify: follow.notify?, languages: follow.languages } }, requested_map: { @account.id => false } }
 
     render json: @account, serializer: REST::RelationshipSerializer, relationships: relationships(**options)
+  rescue FollowService::SelfFollowError
+    render json: { error: I18n.t('Following your own account is not allowed') }, status: 403
   end
```

Comment:
I've opted to use the i18n string here since that fits the pattern used in the codebase. Let me know if there's a specific error class I should have used.